File: rotate_arrays/rotate_arrays.rb
def rotLeft(a, d)
  # Write your code here
  counter = d
  array = a.dup

  while counter.positive?
    element = array.shift
    array << element
    counter -= 1
  end
  array
end

p rotLeft([1, 2, 3, 4, 5], 4)

File: rotate_arrays/rotateArrays.js
function rotLeft(a, d) {
    // Write your code here
    for (let i = d; i > 0; i -= 1) {
        let element = a.shift();
        a.push(element);
    }
    return a;
}

console.log(rotLeft([1, 2, 3, 4, 5], 4))